compiler/src/dotty/tools/dotc/plugins/Plugin.scala

```diff
@@ -28,6 +28,8 @@ sealed trait Plugin {
    *
    *  Research plugin receives a phase plan and return a new phase plan, while
    *  non-research plugin returns a list of phases to be inserted.
+   *
+   *  @note Research plugins only compile with nightly build.
```

Comment:
Do we test this?

Reply:
The logic is enforced with the following property:

```Scala
object Properties {
  val experimental = versionString.contains("SNAPSHOT") || versionString.contains("NIGHTLY")
}
```

We have tests for research plugins. However, non-support of research plugins is only reviewed.